### PR TITLE
[android] Decouple CompassHelper from LocationHelper

### PR DESCRIPTION
**Pull request by rtsisyk**
Compass and location are two different things. Rename SensorsHelper to CompassHelper and extract it from LocationHelper to the upper level to simplify LocationHelper.

Please note that CompassHelper is still started by LocationHelper to avoid any semantic changes.

### CODE REVIEW

File: android/src/app/organicmaps/widget/placepage/PlacePageView.java [hunk 1]
```diff
@@ -265,6 +267,7 @@ public void onStop()
     super.onStop();
     mViewModel.getMapObject().removeObserver(this);
     LocationHelper.INSTANCE.removeListener(this);
+    MwmApplication.from(requireContext()).getCompassHelper().removeListener(this);
```
Comment (rtsisyk) [inline android/src/app/organicmaps/widget/placepage/PlacePageView.java:270]: LocationHelper.INSTANCE and other 11 "enums" can be reworked in the same way.

File: android/src/app/organicmaps/location/CompassHelper.java
Comment (biodranik) [inline android/src/app/organicmaps/location/CompassHelper.java:1]: A more generic Sensor name is more appropriate than Compass IMO. We can later analyze gyro and accelerometer data for more advanced use cases than just "compass".

Comment (rtsisyk) [reply android/src/app/organicmaps/location/CompassHelper.java:1]: Barometer is also a sensor. FaceID is also a sensor... Very generic name. Anyway, I can keep existing SensorHelper name, no problem.

Comment (rtsisyk) [reply android/src/app/organicmaps/location/CompassHelper.java:1]: Renamed.

**Review by biodranik — APPROVED** [review]
LGTM considering all compass-related functionality works as before :)

@Jean-BaptisteC can you please quick-test this PR?

**Review by Jean-BaptisteC — CHANGES_REQUESTED** [review]
❌ Pixel 6 - Android 13

The compass doesn't work on map, when I do rotation arrow doesn't rotate
✅ Compass work on place page

**Comment by rtsisyk** [comment]
> ❌ Pixel 6 - Android 13
> 
> The compass doesn't work on map, when I do rotation arrow doesn't rotate ✅ Compass work on place page

I forgot to wire this new listener to the main activity. Fixed. Thanks!

**Comment by rtsisyk** [comment]
Rebased on the new android project structure.

**Review by Jean-BaptisteC — APPROVED** [review]
✅ Pixel 6 - Android 13

**Pull request merged**